### PR TITLE
feat: Print human readable type name in error messages

### PR DESCRIPTION
**Pull request by l0b0**
Based on <https://stackoverflow.com/a/13477954/96588>.

Also add tests to verify the exception being thrown.

**Related Issue(s):** #331

**Description:**

**PR Checklist:**

- [x] Code is formatted (run `pre-commit run --all-files`)
- [x] Tests pass (run `scripts/test`)
- [ ] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/develop/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

File: pystac/extensions/sar.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def ext(cls, obj: T, add_if_missing: bool = False) -> "SarExtension[T]":
             return cast(SarExtension[T], AssetSarExtension(obj))
         else:
             raise pystac.ExtensionTypeError(
-                f"SAR extension does not apply to type {type(obj)}"
+                f"SAR extension does not apply to type '{type(obj).__name__}'"
```
Comment (duckontheweb) [inline pystac/extensions/sar.py:321]: Thanks for splitting this out into a separate PR. I like this change, and I think for consistency we should apply it to the `ext` methods of the other extensions as well.

Comment (l0b0) [reply pystac/extensions/sar.py]: Done! I've also fixed the names of the extensions in some of them, which all had "File" in the error message.

Comment (duckontheweb) [reply pystac/extensions/sar.py]: Thanks, good catch!

**Review by duckontheweb — APPROVED** [review]
👍

**Pull request merged**